content/master/concepts/composition-functions.md

```diff
@@ -1,1026 +1,699 @@
-    # An arbitrary Kubernetes resource. Passed to the Function as the config
-    # block of its FunctionIO. Doesn't need to exist as a Custom Resource (CR),
-    # since this resource doesn't exist by itself in the API server but must be
-    # a valid Kubernetes resource (have an apiVersion and kind).
-    config:
-      apiVersion: database.example.org/v1alpha1
-      kind: Config
-      metadata:
-        name: cloudsql
-      spec:
-        version: POSTGRES_9_6
+      - name: storage-bucket
+        base:
+          apiVersion: s3.aws.upbound.io/v1beta1
+          kind: Bucket
+        patches:
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.bucketRegion
+          toFieldPath: spec.forProvider.region
 ```
-{{< /expand >}}
-
-Use `kubectl describe <xr-kind> <xr-name>` to debug Composition Functions. Look
-for status conditions and events. Most Functions will emit events associated
-with the XR if they experience issues.
-
-## Building a function
-
- Crossplane doesn't have opinions about how a Composition Function is
- implemented. Functions must:
-
- * Be packaged as an OCI image, where the `ENTRYPOINT` is the Function.
- * Accept input in the form of a `FunctionIO` document on stdin.
- * Return the `FunctionIO` they were passed, optionally mutated, on stdout.
- * Run within the constraints specified by the Composition that includes them,
-   such as timeouts, compute, network access.
 
-This means Functions may be written using a general-purpose programming language
-like Python, Go, or TypeScript. They may also be implemented using a shell
-script, or an existing tool like Helm or Kustomize.
+The `functions.yaml` file contains the Functions the Composition references in
+its pipeline steps:
 
-### FunctionIO
-
-When a Composition Function runner like `xfn` runs your Function it will write
-`FunctionIO` to its stdin. A `FunctionIO` is a Kubernetes style YAML manifest.
-It's not a custom resource (it never gets created in the API server) but it
-follows Kubernetes conventions.
-
-A `FunctionIO` consists of:
-
-* An optional, arbitrary `config` object.
-* The `observed` state of the XR, any existing composed resources, and their connection details.
-* The `desired` state of the XR and any composed resources.
-* Optional `results` of the Function pipeline.
+```yaml
+---
+apiVersion: pkg.crossplane.io/v1beta1
+kind: Function
+metadata:
+  name: function-patch-and-transform
+spec:
+  package: xpkg.upbound.io/crossplane-contrib/function-patch-and-transform:v0.1.4
+```
+{{</expand>}}
+
+The Crossplane CLI uses Docker Engine to run functions. You can change how the
+Crossplane CLI run a function by adding an annotation in `functions.yaml`. Add
+the `render.crossplane.io/runtime` annotation to a Function to change how it's
+run.
+
+`crossplane beta render` supports two `render.crossplane.io/runtime` values:
+
+* `Docker` (the default) connects to Docker Engine. It uses Docker to pull and
+  run function runtimes.
```

Comment:
```suggestion
  run a function runtime.
```

vale fix

Reply:
Hrm weird, I didn't notice this warning.